### PR TITLE
Global AirloopHVACDOAS

### PR DESCRIPTION
**Pull request by mitchute**
### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

**Comment by mitchute** [comment]
Ready for review here.

File: src/EnergyPlus/AirLoopHVACDOAS.cc [hunk 1]
```diff
@@ -87,59 +83,19 @@
 #include <string> // std::string, std::to_string
 
 namespace EnergyPlus {
-namespace AirLoopHVACDOAS {
-
-    int numAirLoopDOAS(0);
```
Comment (mitchute) [inline src/EnergyPlus/AirLoopHVACDOAS.cc:92]: State data members moved to data class

File: src/EnergyPlus/AirLoopHVACDOAS.cc [hunk 2]
```diff
@@ -87,59 +83,19 @@
...
-
-    bool GetInputOnceFlag(true);
-    bool getAirLoopMixerInputOnceFlag(true);
-    bool getAirLoopSplitterInputOnceFlag(true);
-
-    std::vector<AirLoopDOAS> airloopDOAS;
-    std::vector<AirLoopMixer> airloopMixer;
-    std::vector<AirLoopSplitter> airloopSplitter;
-
-    AirLoopMixer::AirLoopMixer() : numOfInletNodes(0), m_AirLoopMixer_Num(0), OutletNodeNum(0), OutletTemp(0.0)
```
Comment (mitchute) [inline src/EnergyPlus/AirLoopHVACDOAS.cc:102]: Constructors moved to the struct declaration in header

File: src/EnergyPlus/AirLoopHVACDOAS.cc [hunk 3]
```diff
@@ -87,59 +83,19 @@
...
-        numAirLoopDOAS = 0;
-    }
+namespace AirLoopHVACDOAS {
 
     void AirLoopDOAS::SimAirLoopHVACDOAS(EnergyPlusData &state, bool const FirstHVACIteration, int &CompIndex)
     {
 
         // Obtains and Allocates unitary system related parameters from input file
-        if (GetInputOnceFlag) {
+        if (dataAirLoopHVACDOAS.GetInputOnceFlag) {
```
Comment (mitchute) [inline src/EnergyPlus/AirLoopHVACDOAS.cc]: References to the state data class

File: src/EnergyPlus/AirLoopHVACDOAS.cc [hunk 4]
```diff
@@ -246,22 +202,22 @@ namespace AirLoopHVACDOAS {
 
     void AirLoopMixer::CalcAirLoopMixer()
     {
-        Real64 OutletTemp = 0.0;
-        Real64 OutletHumRat = 0.0;
-        Real64 MassSum = 0.0;
+        Real64 outletTemp = 0.0;
```
Comment (mitchute) [inline src/EnergyPlus/AirLoopHVACDOAS.cc:203]: Local varaible naming conflicting with member variables

File: src/EnergyPlus/AirLoopHVACDOAS.cc [hunk 5]
```diff
@@ -970,33 +926,33 @@ namespace AirLoopHVACDOAS {
 
     void AirLoopDOAS::SizingAirLoopDOAS(EnergyPlusData &state)
     {
-        Real64 SizingMassFlow = 0;
+        Real64 sizingMassFlow = 0;
```
Comment (mitchute) [inline src/EnergyPlus/AirLoopHVACDOAS.cc:921]: More local variable naming conflicts

File: src/EnergyPlus/AirLoopHVACDOAS.hh [hunk 6]
```diff
@@ -49,72 +49,66 @@
...
 
-    extern int numAirLoopDOAS;
+namespace AirLoopHVACDOAS {
 
     void CheckConvergence();
 
     struct AirLoopMixer
     {
-
-    public:
```
Comment (mitchute) [inline src/EnergyPlus/AirLoopHVACDOAS.hh:69]: Struct members are already public. No need for this.

File: src/EnergyPlus/AirLoopHVACDOAS.hh [hunk 7]
```diff
@@ -49,72 +49,66 @@
...
-        AirLoopMixer(); // constructor
-        ~AirLoopMixer() // destructor
-        {
-        }
-
         std::string name;
-        static AirLoopMixer *factory(int object_type_of_num, std::string const objectName);
+        static AirLoopMixer *factory(int object_type_of_num, std::string const &objectName);
```
Comment (mitchute) [inline src/EnergyPlus/AirLoopHVACDOAS.hh]: Change due to IDE warning suggesting passing by reference

File: src/EnergyPlus/StateManagement.cc [hunk 8]
```diff
@@ -254,6 +255,7 @@ void EnergyPlus::clearAllStates(OutputFiles &outputFiles)
     CurveManager::clear_state();
     AirflowNetwork::clear_state();
     dataAirflowNetworkBalanceManager.clear_state();
+    dataAirLoopHVACDOAS.clear_state();
```
Comment (mitchute) [inline src/EnergyPlus/StateManagement.cc]: New ```clear_state()``` for this data. It doesn't appear that the old one was used anywhere.

**Comment by Myoldmopar** [comment]
@mitchute this is looking happy so far.  If CI is completely happy, I think this can go in.

**Comment by mitchute** [comment]
CI is happy with this. I'm gonna merge it.

**Pull request merged**